cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -277,10 +199,11 @@ void run_mnist(const std::string& model_name,
 
   // Begin the actual iteration loop
   log_data(context, LLDebug, "Iteration loop starting...");
+  MPI_Barrier(MPI_COMM_WORLD); // try to put one in between all of them
   double loop_start = MPI_Wtime();
   for (int i = 0; i < iterations + 1; i++) {
     log_data(context, LLDebug, "Running iteration: " + std::to_string(i));
-    MPI_Barrier(MPI_COMM_WORLD);
+    // MPI_Barrier(MPI_COMM_WORLD); // move this to outside loop just as test case
```

Comment:
Resolve where the barrier needs to be. This is messing up your timings

Reply:
moved outside of loop like the other funcs so i am keeping this but deleting the comment